idioms/option-list.md

```diff
@@ -0,0 +1,29 @@
+# Treating `Option` like a list
+
+## Description
+
+`Option` can be viewed as a list that contains either zero or one elements. It can be iterated over using `Option::iter()`, and converted to a slice using the `ref_slice` crate. Keeping these tools in mind can lead to simpler, more idiomatic code.
+
+## Examples
+
+```rust
+let arg = Some("doovalacky");
+// equivalent to `if let Some(value) = ...`
+for value in arg {
+    println!("Received: {}", value);
+}
+```
```

Comment:
A better example might be using `Chain` to append an Option to another iterator, and/or a function that takes `T: Iterator<U>`

I would sat that using a `for` loop with an `Option` is less idiomatic than using `if let` (and it might be worth stating that).